test/utils/session-core.test.js

```diff
@@ -0,0 +1,249 @@
+import {
+  getSolidDataset,
+  createContainerAt,
+  mockSolidDatasetFrom,
+  mockThingFrom,
+  createSolidDataset,
+  setThing,
+  saveSolidDatasetInContainer
+} from '@inrupt/solid-client';
+import { expect, vi, it, describe, afterEach, beforeEach } from 'vitest';
+import { getContainerUrl } from '../../src/utils/network/session-helper';
+import { INTERACTION_TYPES } from '../../src/constants';
+
+vi.mock('@inrupt/solid-client', async () => {
+  const actual = await vi.importActual('@inrupt/solid-client');
+  return {
+    ...actual,
+    getSolidDataset: vi.fn((url) => Promise.resolve(mockSolidDatasetFrom(url))),
+    createContainerAt: vi.fn((url) => Promise.resolve(mockSolidDatasetFrom(url))),
+    saveSolidDatasetInContainer: vi.fn((url) => Promise.resolve(mockSolidDatasetFrom(url)))
+  };
+});
+
+const getSolidDatasetRejected = vi.fn(() => Promise.reject(Error('dataset not found')));
+
+const mockPodUrl = 'https://pod.example.com/';
+const mockSolidIdentityProvider = 'https://example.com/';
+let session = {};
+
+describe('setDocAclPermission', () => {
+  beforeEach(() => {
+    session = {
+      fetch: vi.fn(),
+      info: {
+        webId: `${mockPodUrl}profile/card#me`
+      }
+    };
+  });
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it("prints out correct documentUrl for Driver's License and webId for other user", () => {
+    const fileType = "Driver's License";
+    const otherPodUsername = 'pod2';
+    const documentUrl = getContainerUrl(session, fileType, INTERACTION_TYPES.SELF);
+    const webId = `https://${otherPodUsername}.${
+      mockSolidIdentityProvider.split('/')[2]
+    }/profile/card#me`;
+
+    expect(documentUrl).toBe('https://pod.example.com/Drivers%20License/');
+    expect(webId).toBe('https://pod2.example.com/profile/card#me');
```

Comment:
Not sure what you're testing here. The webId is generated from test data. It doesn't get any data from`getContainerUrl`

Reply:
At the moment the function setDocAclPermission currently uses the other person's username to build a webId. The test here is to check if the generated variables correct because being passed into setDocAclForUser:

```JavaScript
export const setDocAclPermission = async (session, fileType, permissions, otherPodUsername) => {
  const documentUrl = getContainerUrl(session, fileType, INTERACTION_TYPES.SELF);
  const webId = `https://${otherPodUsername}.${
    SOLID_IDENTITY_PROVIDER.split('/')[2]
  }/profile/card#me`;

  await setDocAclForUser(session, documentUrl, 'update', webId, permissions);
};
```

Could add an expect for setDocAclForUser that it's expected to called with expected values